lib/chef/property.rb

```diff
@@ -448,18 +452,22 @@ def emit_dsl
       # stack trace if you use `define_method`.
       declared_in.class_eval <<-EOM, __FILE__, __LINE__+1
         def #{name}(value=NOT_PASSED)
+          raise "Property #{name} of \#{self} cannot be passed a block!" if block_given?
```

Comment:
Since this is going to be hit when the user does the `property :template` and later calls `template "foo" do..end` this error message should probably contain more useful information about what to do in that case.  From the user's perspective this is kinda shitty because they typed a template resource and property is blowing up, and they're likely to WTF around for awhile before the lightbulb goes on...